### PR TITLE
Fix ASTM consumer boundary bugs (sender shape, instrument cascade, sample fallback)

### PR DESCRIPTION
**Pull request by ramonski**
## Description of the issue/feature this PR addresses

Boundary hygiene pass on `senaite.core.astm.{consumer,importer}`. Four small bugs were surfaced by a code review of the subpackage in preparation for upcoming work (1.x↔2.x envelope compatibility, generic profile-driven importer); none of them changes the public adapter API, so customer adapters continue to work unchanged.

The package also accepts `senaite.astm` 1.x envelopes (where `H[0].sender` could be a list `[name, serial, version]` instead of the 2.x dict). Both shapes now route through the same dispatch.

## Current behavior before PR

- `PushConsumer.get_sender_name` has `and` where `or` was meant — a non-list `H` (e.g. `None`, a stray dict, or a 1.x edge case) crashes inside `len(header)` before the default can return.
- `PushConsumer.process`'s per-message exception clause builds an error string into a local named `message` (shadowing the loop variable), never logs it, and logs the exception without a traceback.
- `ASTMImporter.instrument` iterates every Instrument and overwrites the match on every loop pass — a weaker later match (title-only) silently displaces a stronger earlier match (model+serial).
- `ASTMImporter.sample` falls back from `getId` to `getClientSampleID` with no logging and no client scoping. Client Sample IDs are not unique across clients, so a multi-hit fallback silently returns `None`, and a single-hit fallback can import into the wrong client's sample.
- The dispatch consumer name docstring reads `"enaite.core.lis2a.import"` (missing `s`).
- The `README.md` documents an outdated envelope shape and does not mention the 2.x `metadata` block or HL7 transport.

## Desired behavior after PR is merged

- `get_sender_name` returns the default cleanly when `H` is not a one-element list. The sender is accepted as either a dict (`{"name": ..., "serial": ..., "version": ...}` — 2.x) or a list/tuple (`[name, serial, version]` — 1.x).
- `process` uses `logger.exception(...)` (traceback + offending payload) and no longer shadows the loop variable.
- `instrument` ranks candidates (model+serial = 3, title+serial = 2, model = 1, title = 1) and returns the best score; logs when multiple candidates tie at the top.
- `sample` keeps the `getClientSampleID` fallback for backwards compatibility but only accepts a single hit; ambiguous or empty results are logged so the operator sees what happened instead of getting a silent `None`.
- `README.md` rewritten: documents the current 2.x envelope (with `metadata.envelope_version`, raw payload in `metadata.{astm,lis2a,hl7}`, the H/P/O/R/C/M/L/Q bucket layout shared with HL7), the named-adapter dispatch, and the importer API.
- New unit test `senaite/core/tests/test_astm_consumer.py` covers `get_sender_name` across both envelope shapes plus empty/missing/multi-header/no-name edge cases.

### CODE REVIEW

File: src/senaite/core/astm/importer.py [hunk 1]
```diff
@@ -54,61 +55,116 @@ def import_data(self):
...
+    @staticmethod
+    def _score_instrument_candidate(obj, name, serial):
+        model = obj.getModel()
+        title = obj.Title()
+        serialno = obj.getSerialNo()
+        if not name:
+            return 0
+        if model == name and serialno == serial:
+            return 3
+        if title == name and serialno == serial:
```
Comment (xispa) [inline src/senaite/core/astm/importer.py:112]: I would force stripped and lowercase comparisons. I've found cases where the instance's title is e.g. "Genexpert", while the machine states "GeneXpert". Besides, quite often lab personnel introduce trailing or leading whitespaces because of copy&paste without noticing (this is the main reason why we added our own `TextLineField` with https://github.com/senaite/senaite.core/pull/2791 )

**Review by xispa — APPROVED** [review]
Just left a comment related with comparisons. However, not part of this PR, as the comparisons hasn't changed, but the weights of matches to get the best instrument candidate.
